lib/shared/bucketing/__tests__/bucketing.test.ts

```diff
@@ -1415,6 +1415,118 @@ describe('Rollout Logic', () => {
                 doesUserPassRollout({ rollout, boundedHash: 0.9 }),
             ).toBeFalsy()
         })
+
+        it('should handle stepped rollout with 0% start and 100% later stage', () => {
+            const rollout: PublicRollout = {
+                type: 'stepped',
+                startDate: new Date(),
+                startPercentage: 0,
+                stages: [
+                    {
+                        type: 'discrete',
+                        date: new Date(
+                            new Date().getTime() + 1000 * 60 * 60 * 24 * 7,
+                        ),
+                        percentage: 1,
+                    },
+                ],
+            }
+
+            // Before next stage - should be 0%
+            jest.useFakeTimers().setSystemTime(new Date())
+            for (let i = 0; i < 100; i++) {
+                expect(
+                    doesUserPassRollout({ rollout, boundedHash: i / 100 }),
+                ).toBeFalsy()
+            }
+
+            // After 100% stage - should pass all users
+            jest.useFakeTimers().setSystemTime(
+                new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 8),
+            )
+            for (let i = 0; i < 100; i++) {
+                expect(
+                    doesUserPassRollout({ rollout, boundedHash: i / 100 }),
+                ).toBeTruthy()
+            }
+
+            jest.useRealTimers()
+        })
+
+        it('should handle stepped rollout with 50% start and 100% later stage', () => {
+            const rollout: PublicRollout = {
+                type: 'stepped',
+                startDate: new Date(),
+                startPercentage: 0.5,
+                stages: [
+                    {
+                        type: 'discrete',
+                        date: new Date(
+                            new Date().getTime() + 1000 * 60 * 60 * 24 * 7,
+                        ),
+                        percentage: 1,
+                    },
+                ],
+            }
+
+            // Before next stage - should be 50%
+            jest.useFakeTimers().setSystemTime(new Date())
+            expect(
+                doesUserPassRollout({ rollout, boundedHash: 0.49 }),
```

Comment:
based on my understanding of how stepped rollouts are supposed to work, this test is wrong

Reply:
what's wrong with it?

Reply:
see slack